internal/annotations/prefix.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotations
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	SDKPrefix = "+operator-sdk"
+
+	prefixSep = ":"
+	pathSep   = "."
+	valueSep  = "="
+)
+
+func JoinPrefix(tokens ...string) string {
+	return strings.Join(tokens, prefixSep)
+}
+
+func SplitPrefix(prefix string) ([]string, error) {
+	if !strings.Contains(prefix, prefixSep) {
+		return nil, fmt.Errorf(`prefix "%s" does not contain the prefix separator "%s"`, prefix, prefixSep)
+	}
+	split := strings.Split(prefix, prefixSep)
+	if len(split) == 0 || (len(split) == 1 && split[0] == "") {
+		return nil, fmt.Errorf(`prefix "%s" has no prefix tokens delimited by "%s"`, prefix, prefixSep)
+	}
+	if strings.TrimSpace(split[0]) != SDKPrefix {
+		return nil, fmt.Errorf(`prefix "%s" does not have SDK prefix "%s"`, prefix, prefixSep)
+	}
+	for i, p := range split {
+		if strings.TrimSpace(p) == "" {
+			return nil, fmt.Errorf(`prefix "%s" contains an empty token after colon %d`, prefix, i)
```

Comment:
i is index starting from 0.

So the following error message might be slightly ambiguous:
```console
... contains an empty token after colon 1
```
Is that the 1st colon?

So either add the word index, or use `i+1`:
```console
contains an empty token after colon index 1
```
```suggestion
			return nil, fmt.Errorf(`prefix "%s" contains an empty token after colon index %d`, prefix, i)
```